src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx

```diff
@@ -0,0 +1,336 @@
+import React from 'react'
+
+import { ClaimStatus } from 'audius-client/src/common/store/pages/audio-rewards/slice'
+import { StyleSheet, View, ImageSourcePropType } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+
+import IconCheck from 'app/assets/images/iconCheck.svg'
+import Button, { ButtonType } from 'app/components/button'
+import Drawer from 'app/components/drawer'
+import GradientText from 'app/components/gradient-text'
+import LoadingSpinner from 'app/components/loading-spinner'
+import Text from 'app/components/text'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { ThemeColors, useThemeColors } from 'app/utils/theme'
+
+const messages = {
+  task: 'Task',
+  reward: 'Reward',
+  progress: 'Progress',
+  audio: '$AUDIO',
+  incomplete: 'Incomplete',
+  complete: 'Complete',
+  claim: 'Claim Your Reward',
+  claimErrorMessage: 'Oops, something’s gone wrong'
+}
+
+const createProgressBarStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    progressBarContainer: {
+      backgroundColor: themeColors.neutralLight9,
+      borderRadius: 22,
+      height: 24,
+      marginTop: 14,
+      marginBottom: 14,
```

Comment:
No need to update, but I recently learned that you can use marginVertical for things like this

Reply:
wow!